faviconupdate.js

```diff
@@ -1,165 +1,230 @@
-	"kinesis":						{href: "kinesis.png"},
-	//"lambda":						{href: "lambda.png"},							// Amazon did this one!
-	"lex":								{href: "lex.png"},
-	"ls":									{href: "ls.png"},									// Reminder: This ls means LightSail
-	//"macie":						{href: "macie.png"},							// Needs special handling for unique URL
-	"machinelearning":		{href: "machinelearning.png"},
-	"managed-services":		{href: "managed-services.png"},
-	"mediaconvert":				{href: "mediaconvert.png"},
-	"medialive":					{href: "medialive.png"},
-	"migrationhub":				{href: "migrationhub.png"},				// This used to be Discovery
-	"mobilehub":					{href: "mobilehub.png"},
-	"opsworks":						{href: "opsworks.png"},
-	"pinpoint":						{href: "pinpoint.png"},
-	"polly":							{href: "polly.png"},
-	"rds":								{href: "rds.png"},
-	"redshift":						{href: "redshift.png"},
-	"rekognition":				{href: "rekognition.png"},
-	"route53":						{href: "route53.png"},
-	"s3":									{href: "s3.png"},
-	"servermigration":		{href: "servermigration.png"},
-	"servicecatalog":			{href: "servicecatalog.png"},
-	"ses":								{href: "ses.png"},
-	"sns":								{href: "sns.png"},
-	"sqs":								{href: "sqs.png"},
-	"states":							{href: "states.png"},
-	"storagegateway":			{href: "storagegateway.png"},
-	"trustedadvisor":			{href: "trustedadvisor.png"},
-	"swf": 								{href: "swf.png"},								// Needs favicon
-	"waf": 								{href: "waf.png"},
-	"workspaces": 				{href: "workspaces.png"},					// This one is bugged for some unknown reason (see if statement below)
-	"workmail": 					{href: "workmail.png"},
-	"vpc": 								{href: "vpc.png"},
-	"xray": 							{href: "xray.png"},
-	"zocalo": 						{href: "zocalo.png"},
-	
-}
+/**
+ * To pull all the icons and their positions automatically from the Resource Groups tab run this in the console
+ **/
+// var icons = {};
+// document.querySelectorAll('span.service-icon.service-icon-navbar.service-icon-size-med').forEach(item => {
+//     let href = item.parentElement.parentElement.parentElement.getAttribute('data-service-href');
+//     if (!item) return;
+//     let reg = /:\/\/([a-z0-9.-]*)\/([a-z0-9.-]*)\/([a-z0-9.-]*)\/*/g;
+//
+// // Break the URL into parts and capture the string after the 'amazon.com/' as awsServiceName
+//     let captureGroupArray = Array.from(href.matchAll(reg));
+//
+//     if (!captureGroupArray || !captureGroupArray.length || captureGroupArray[0].length < 3) return;
+//     let awsServiceName = captureGroupArray[0][2];
+//
+// // For Codesuite URLs (Codebuild/CodeDeploy/CodePipeline/etc...), we need to break the URL apart further to work
+//     if (awsServiceName === 'codesuite') {
+//         awsServiceName = captureGroupArray[0][3];
+//     }
+//     icons[awsServiceName] = item.style.backgroundPosition.replace(/[^0-9 ]/g, '').split(' ').map(x => +x);
+//
+// });
+// JSON.stringify(icons);
 
+// Define all the AWS services here
+//How to get all these icon positions
+const ICONS = {
+    "iam": [448, 24],

```

Comment:
Thoughts on having the list in the order produced by the console script?

Could make future code reviews easier

We could modify the console script to sort and add whitespace / new lines first

Reply:
The new changes load all the services automatically so there is no need to create that list at all.